azure-cli/ad/deploy.swagger.ps1

```diff
@@ -0,0 +1,80 @@
+param (
+  [Parameter(Mandatory = $true)]
+  [ValidateNotNullOrEmpty()]
+  [string]
+  $companyHostName,
+
+  [Parameter(Mandatory = $true)]
+  [EnvironmentConfig] $environmentConfig,
+
+  [Parameter(Mandatory = $true)]
+  [NamingConfig] $namingConfig,
+
+  [Parameter(Mandatory = $false)]
+  [ValidateNotNullOrEmpty()]
+  [string]
+  $serviceInstance = ""
+)
+
+# import utility functions
+. "$PSScriptRoot\..\utilities\deploy.naming.ps1"
+
+$spnAppIdentityId = Get-AppIdentityUri `
+  -type "api" `
+  -companyHostName $companyHostName `
+  -environmentConfig $environmentConfig `
+  -namingConfig $namingConfig `
+  -serviceInstance $serviceInstance
+
+$spnAppIdentityName = Get-AppIdentityDisplayName `
+  -type "spn" `
+  -environmentConfig $environmentConfig `
+  -namingConfig $namingConfig `
+  -serviceInstance "swagger"
+
+$appId = az ad app list `
+  --identifier-uri $spnAppIdentityId `
+  --query [0].appId
+
+Write-Host "Creating Swagger App Registration" -ForegroundColor DarkGreen
+$clientId = az ad app create `
+  --display-name $spnAppIdentityName `
+  --oauth2-allow-implicit-flow true `
+  --query appId
+
+$objectId = az ad sp show --id $clientId --query objectId
+
+if (!$objectId) {
+  Write-Host "  Creating Service Principal for Swagger App Registration" -ForegroundColor DarkGreen
+
+  $objectId = az ad sp create --id $clientId --query objectId
+}
+
+Write-Host "  Granting Swagger SPN access to App" -ForegroundColor DarkGreen
+az ad app permission grant --id $clientId --api $appId
+Start-Sleep -Seconds 30 # This ARBITRARY delay time is required otherwise next call to grant admin consent will fail (SOMETIMES!)
+
+Write-Host "  Giving Admin permission" -ForegroundColor DarkGreen
+az ad app permission admin-consent --id $clientId
+
+Write-Host "  Setting redirect URIs" -ForegroundColor DarkGreen
+$swaggerAppObjectId = az ad app show --id $clientId --query objectId
+
+$redirectUri = "https://$($namingConfig.ServiceAbbreviation).$($environmentConfig.EnvironmentName).$companyHostName/swagger/oauth2-redirect.html"
+
+$redirectUris = "[\""$($redirectUri.ToLower())\""]"
+
+if ($environmentConfig.environmentType -ne "Production") {
+  $redirectUris = "[\""$($redirectUri.ToLower())\"", \""https://localhost:5001/swagger/oauth2-redirect.html\""]"
```

Comment:
This line looks sketchy. Can you share the intended result ?

Reply:
The result of this is that localhost is added as a redirctUri. I think this is needed for making it work when running the web api locally but I am not sure. So I will do a bit more investigation.

Reply:
I'm talking about the JSON string you construct. It's unreadable with the double quotes and backslashes. Can't tell if it's supposed to be escaped JSON or not.
If it's not escaped JSON you can write it like
```
$redirectUri = "github.com"
$redirectUris = @($redirectUri, "https://localhost:5001/swagger/oauth2-redirect.html") | ConvertTo-Json -Compress
```
`$redirectUris` will then be a string with `["github.com","https://localhost:5001/swagger/oauth2-redirect.html"]` as value